### PR TITLE
[New Rule] Persistence via Cron Tasks

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/866

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+[metadata]
+creation_date = "2021/01/15"
+maturity = "production"
+updated_date = "2021/01/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or the execution of a cron task. Adversaries may abuse the cron jobs to perform task scheduling
+for initial or recurring execution of malicious code.
```
Comment (peasead) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:10]:
```suggestion
Identifies the creation, or execution of, a cron task. 
```

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+"""
+false_positives = ["Legit software or scripts using cron jobs for recurrent tasks execution."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Persistence via Cron Tasks on macOS"
```
Comment (peasead) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:17]:
```suggestion
name = "Persistence via Cron Tasks on macOS"
note = "Adversaries may abuse the cron jobs to perform task scheduling for initial or recurring execution of malicious code."
```

Comment (peasead) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:17]:
```suggestion
name = "Persistence via cron Tasks on macOS"
```

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+references = ["https://archive.f-secure.com/weblog/archives/00002576.html", "https://ss64.com/osx/crontab.html"]
+risk_score = 47
+rule_id = "b1c14366-f4f8-49a0-bcbb-51d2de8b0bb8"
+severity = "medium"
```
Comment (peasead) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:21]:
```suggestion
severity = "low"
```

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+for initial or recurring execution of malicious code.
+"""
+false_positives = ["Legit software or scripts using cron jobs for recurrent tasks execution."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Persistence via Cron Tasks on macOS"
+references = ["https://archive.f-secure.com/weblog/archives/00002576.html", "https://ss64.com/osx/crontab.html"]
+risk_score = 47
```
Comment (peasead) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:19]:
```suggestion
risk_score = 21
```

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+[metadata]
+creation_date = "2021/01/15"
+maturity = "production"
+updated_date = "2021/01/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation, or execution of, a cron task.
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:9]:
```suggestion
Identifies the creation or execution of a cron job. Adversaries may abuse the cron jobs to perform task scheduling for initial or recurring execution of malicious code.
```

Comment (bm11100) [reply rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:9]: re lint

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+description = """
+Identifies the creation, or execution of, a cron task. 
+"""
+false_positives = ["Legit software or scripts using cron jobs for recurrent tasks execution."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Persistence via cron Tasks on macOS"
+note = "Adversaries may abuse the cron jobs to perform task scheduling for initial or recurring execution of malicious code."
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:17]:
```suggestion

```
Should be part of the description.

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+author = ["Elastic"]
+description = """
+Identifies the creation, or execution of, a cron task. 
+"""
+false_positives = ["Legit software or scripts using cron jobs for recurrent tasks execution."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Persistence via cron Tasks on macOS"
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:16]:
```suggestion
name = "Persistence via Cron Job"
```

**Comment by bm11100** [comment]
Do we think this one may be a bit noisy based on testing?

**Comment by Samirbous** [comment]
> Do we think this one may be a bit noisy based on testing?

@bm11100 its an important technique to cover, noise can't be avoided for this kind of events but we've already excluded root user and some other executable paths. up to the analyst to tune per their environment.

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 8]
```diff
@@ -0,0 +1,57 @@
+[metadata]
+creation_date = "2021/01/15"
+maturity = "production"
+updated_date = "2021/01/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or execution of a cron job. Adversaries may abuse the cron jobs to perform task scheduling for initial or recurring execution of malicious code.
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:9]: re lint

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+"""
+false_positives = ["Legit software or scripts using cron jobs for recurrent tasks execution."]
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:11]:
```suggestion
false_positives = ["Legitimate software or scripts using cron jobs for recurring tasks."]
```

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+   (process.name:crontab and not process.args:("-l" or "-r" or "-e" or "-help" or "-h") ) or 
+   (process.parent.name:cron and not process.name:"running job" and
+    not process.executable:
+                 (/Applications/Docker.app/Contents/Resources/bin/docker or 
+                  /usr/bin/killall or 
+                  /usr/sbin/sendmail or 
+                  /usr/bin/env or 
+                  /usr/bin/timeshift or
+                  /bin/rm))
+  )
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:37]:
```suggestion
event.category:process and event.type:(start or process_started or info) and
  not user.name:root and
  (
    (process.name:crontab and not process.args:("-l" or "-r" or "-e" or "-help" or "-h")) or
    (process.parent.name:cron and not process.name:"running job" and
      not process.executable:
      (
        /Applications/Docker.app/Contents/Resources/bin/docker or
        /usr/bin/killall or
        /usr/sbin/sendmail or
        /usr/bin/env or
        /usr/bin/timeshift or
        /bin/rm
      )
    )
  )
```

Comment (bm11100) [reply rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:37]: spacing

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 11]
```diff
@@ -0,0 +1,62 @@
+[metadata]
+creation_date = "2021/01/15"
+maturity = "production"
+updated_date = "2021/01/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or execution of a cron job. Adversaries may abuse the cron jobs to perform task scheduling for
+initial or recurring execution of malicious code.
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:10]:
```suggestion
Identifies the creation or execution of a cron job. Adversaries may abuse cron jobs to perform task scheduling for
initial or recurring execution of malicious code.
```

File: rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+"""
+false_positives = ["Legitimate software or scripts using cron jobs for recurring tasks."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Persistence via Cron Job"
```
Comment (bm11100) [inline rules/cross-platform/persistence_cron_jobs_creation_and_runtime.toml:17]:
```suggestion
name = "Potential Persistence via Cron Job"
```

**Review by bm11100 — APPROVED** [review]

**Review by brokensound77 — APPROVED** [review]
I liked your initial approach, targeting the cron files which hold the job info (though we would need to expand). I think we can see how this does, and if need be, move to a file based rule

**Comment by Samirbous** [comment]
> I liked your initial approach, targeting the cron files which hold the job info (though we would need to expand). I think we can see how this does, and if need be, move to a file based rule

the current rule looks for both creation state (crontab cmdline) and runtime-state (child of cron), the filemod approach won't be useful to spot the content of the job but for sure good to complement.

**Pull request merged**